### PR TITLE
Fix indentation

### PR DESCRIPTION
**Pull request by conao3**
Fix indentation.
For #2689 discussion, we should have some indentation lint to
enforce the indentation style (using Emacs-28).

This PR prerequired to introduce that linter.
```sh
$ emacs --version
GNU Emacs 28.0.50
Copyright (C) 2021 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

$ ls -1 | grep \.el | grep -v \.dir-locals\.el | xargs -t -I^ sh -c "cask emacs -q --batch ^ --eval \"(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))\" || true"
sh -c 'cask emacs -q --batch lsp-completion.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-diagnostics.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-dired.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-headerline.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-icons.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-ido.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
^[[OIndenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-iedit.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-lens.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-mode.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-modeline.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-protocol.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done
sh -c 'cask emacs -q --batch lsp-semantic-tokens.el --eval "(let ((require-final-newline t)) (eval-buffer) (untabify (point-min) (point-max)) (indent-region (point-min) (point-max)) (save-buffer))" || true'
Indenting region... 
Indenting region...done`
```

### CODE REVIEW

**Comment by conao3** [comment]
Sorry, something wrong this patch

**Comment by conao3** [comment]
Re-check this patch, this is good. GitHub diff algorithm bit confused.

File: lsp-diagnostics.el [hunk 1]
```diff
@@ -115,12 +115,12 @@ g. `error', `warning') and list of LSP TAGS."
                (bitmap (or (get flycheck-level 'flycheck-fringe-bitmaps)
                            (get flycheck-level 'flycheck-fringe-bitmap-double-arrow))))
           (flycheck-define-error-level new-level
```
Comment (yyoncho) [inline lsp-diagnostics.el:117]: lets format that with flycheck loaded

File: lsp-diagnostics.el [hunk 2]
```diff
@@ -205,15 +205,15 @@ from the language server."
...
+                                     "A syntax checker using the Language Server Protocol (LSP)
 provided by lsp-mode.
 See https://github.com/emacs-lsp/lsp-mode."
-      :start #'lsp-diagnostics--flycheck-start
-      :modes '(lsp-placeholder-mode) ;; placeholder
-      :predicate (lambda () lsp-diagnostics-mode)
-      :error-explainer (lambda (e)
-                         (lsp-diagnostics-flycheck-error-explainer
-                          e (lsp--workspace-server-id (car-safe (lsp-workspaces))))))))
+                                     :start #'lsp-diagnostics--flycheck-start
```
Comment (yyoncho) [inline lsp-diagnostics.el:211]: ditto

File: lsp-diagnostics.el (hunk 1 above)
Comment (conao3) [reply lsp-diagnostics.el:117]: If i add `(require 'flycheck)` on top of file. Can I?

File: lsp-completion.el [hunk 3]
```diff
@@ -213,16 +213,16 @@ Return `nil' when fails to guess prefix."
...
-                 :item ,item
-                 :start-point ,start-point
-                 :score ,score?))
-             it)))
+    (-map (-lambda ((item &as &CompletionItem
+                          :label
+                          :filter-text?
+                          :_emacsStartPoint start-point
+                          :score?))
+            `(:label ,(or filter-text? label)
```
Comment (yyoncho) [inline lsp-completion.el:221]: I think the agreement is to have plist on the same level, lets reformat that once @leungbk implements dev formatting as he mentioned in https://github.com/emacs-lsp/lsp-mode/pull/2689#issuecomment-792293282

File: lsp-diagnostics.el (hunk 1 above)
Comment (yyoncho) [reply lsp-diagnostics.el:117]: flycheck is an optional dependency but at the same time, most of the devs work with flycheck already loaded.

Comment (conao3) [reply lsp-diagnostics.el:117]: ok, I will add `flycheck` as a cask `dev-dependency`.

File: lsp-completion.el (hunk 3 above)
Comment (leungbk) [reply lsp-completion.el:221]: I've moved the formatting stuff to a separate mode and I might need some time to iron out the kinks. I'm fine with leaving the indentation like this until I have that mode ready.

File: lsp-completion.el [hunk 4]
```diff
@@ -213,16 +213,16 @@ Return `nil' when fails to guess prefix."
...
-                             :label
-                             :filter-text?
-                             :_emacsStartPoint start-point
-                             :score?))
-               `(:label ,(or filter-text? label)
-                 :item ,item
-                 :start-point ,start-point
-                 :score ,score?))
-             it)))
+    (-map (-lambda ((item &as &CompletionItem
```
Comment (kiennq) [inline lsp-completion.el:216]: The indentation change for `-->` is recent with https://github.com/magnars/dash.el/commit/92562909a763d766fd81b817418c2e05d35d50e0

File: lsp-completion.el [hunk 5]
```diff
@@ -465,7 +465,7 @@ The MARKERS and PREFIX value will be attached to each candidate."
           ;; metadata
           ((equal action 'metadata)
            `(metadata (category . lsp-capf)
-             (display-sort-function . identity)))
+                      (display-sort-function . identity)))
```
Comment (kiennq) [inline lsp-completion.el:468]: This indentation is intended as a plist. You should not change this or add `( metadata` so it will be aligned

Comment (conao3) [reply lsp-completion.el:468]: This is not `plist`.
Maybe you want this?
```diff
          (cond
           ;; metadata
           ((equal action 'metadata)
-           `(metadata (category . lsp-capf)
-                      (display-sort-function . identity)))
+           `(metadata
+             (category . lsp-capf)
+             (display-sort-function . identity)))
```

**Review by yyoncho — APPROVED** [review]
Looks good to me. (good to go after addressing @kiennq comments)

File: lsp-completion.el (hunk 3 above)
Comment (yyoncho) [reply lsp-completion.el:221]: @leungbk - given the fact that it seems like there is only one place that is affected apparently after lsp-protocol.el migration I wonder whether it makes sense to invest effort in that. WDYT?

File: lsp-completion.el (hunk 5 above)
Comment (kiennq) [reply lsp-completion.el:468]: Ah, nvm, the change on this looks good with 
``` elisp
           `(metadata (category . lsp-capf)
                      (display-sort-function . identity)))
```

Comment (conao3) [reply lsp-completion.el:468]: ? I cannot understand `nvm` but, finally, you approve this?

File: lsp-completion.el (hunk 3 above)
Comment (kiennq) [inline lsp-completion.el:221]: Can you make this 
```suggestion
            `( :label ,(or filter-text? label)
```
so the subsequent item will align nicely?

Comment (conao3) [reply lsp-completion.el]: approve your suggestion.

File: lsp-completion.el (hunk 5 above)
Comment (kiennq) [reply lsp-completion.el:468]: `nvm` is stand for `nevermind`.

Comment (conao3) [reply lsp-completion.el:468]: Thanks, i memorize it.

**Review by kiennq — APPROVED** [review]

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**